tiledb/sm/filesystem/posix.cc

```diff
@@ -137,16 +137,17 @@ Posix::Posix(const Config& config) {
 void Posix::create_dir(const URI& uri) const {
   // If the directory does not exist, create it
   auto path = uri.to_path();
-  if (is_dir(uri)) {
-    throw IOError(
-        std::string("Cannot create directory '") + path +
-        "'; Directory already exists");
-  }
-
-  if (mkdir(path.c_str(), directory_permissions_) != 0) {
+  throw_if_not_ok(ensure_directory(path));
+  auto status = mkdir(path.c_str(), directory_permissions_);
+  if (status != 0) {
+    auto err = errno;
+    if (err == EEXIST) {
+      // Do not fail if directory already existed.
```

Comment:
This is the documented behavior, so I'm treating this change as a bug fix.

https://github.com/TileDB-Inc/TileDB/blob/639322fbcf068dad54b4906078a8a4cfb381d70e/tiledb/api/c_api/vfs/vfs_api_external.h#L229-L230